numpy/core/fromnumeric.py

```diff
@@ -1191,16 +1197,28 @@ def argmax(a, axis=None, out=None):
     >>> np.take_along_axis(x, np.expand_dims(index_array, axis=-1), axis=-1).squeeze(axis=-1)
     array([4, 3])
 
+    Setting `keepdims` to `True`,
+
+    >>> x = np.arange(24).reshape((2, 3, 4))
+    >>> res = np.argmax(x, axis=1, keepdims=True)
+    >>> res.shape
+    (2, 1, 4)
     """
-    return _wrapfunc(a, 'argmax', axis=axis, out=out)
+    res = _wrapfunc(a, 'argmax', axis=axis, out=out)
+    if keepdims:
+        curr_shape = a.shape
+        new_shape = curr_shape[:axis] + (1,) + curr_shape[axis + 1:]
```

Comment:
Does this work for `axis=None, keepdims=True`? It looks like it will end with:
```
    ... curr_shape[1 + axis:]
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
```
Comparing with `max`:
```python
>>> np.max(x, axis=1, keepdims=True).shape
(2, 1, 4)
>>> np.max(x, axis=None, keepdims=True).shape
(1, 1, 1)
```